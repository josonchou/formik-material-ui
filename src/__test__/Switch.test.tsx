import * as React from 'react';
import { Formik, Field, Form } from 'formik';
import renderer from 'react-test-renderer';

import { Switch } from '../Switch';

test('Switch Renders Correctly', () => {
  const component = renderer.create(
    <Formik initialValues={{}} onSubmit={() => null}>
      <Form>
        <Field name="test" label="Text" component={Switch} />
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
