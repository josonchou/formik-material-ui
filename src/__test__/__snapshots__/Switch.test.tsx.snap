// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Renders Correctly 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <span
    className="MuiSwitch-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-12 MuiSwitch-switchBase MuiSwitch-colorSecondary"
      label="Text"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          className="MuiSwitch-thumb"
        />
        <input
          className="PrivateSwitchBase-input-15 MuiSwitch-input"
          disabled={false}
          name="test"
          onChange={[Function]}
          type="checkbox"
          value="test"
        />
      </span>
    </span>
    <span
      className="MuiSwitch-track"
    />
  </span>
</form>
`;
